bluemix/configuration/persistor.go

```diff
@@ -44,14 +47,26 @@ func (dp DiskPersistor) Load(data DataInterface) error {
 		return err
 	}
 
-	if err != nil {
-		err = dp.write(data)
+	if err != nil { // strange: requiring an error (to allow write attempt to continue), as long as it is not a permission error
+		err = dp.lockedWrite(data)
 	}
 	return err
 }
 
+func (dp DiskPersistor) lockedWrite(data DataInterface) error {
+	lockErr := dp.fileLock.Lock() // provide a file lock, in addition to the RW mutex (in calling functions), just while dp.write is called
```

Comment:
Nit: For readability can we move the comments above line 57

Reply:
Sure!